moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -709,14 +707,14 @@ bool ompl_interface::ModelBasedPlanningContext::solve(planning_interface::Motion
   else
   {
     ROS_INFO_NAMED(LOGNAME, "Unable to solve the planning problem");
-    res.error_code_.val = moveit_msgs::MoveItErrorCodes::PLANNING_FAILED;
     return false;
   }
 }
 
 bool ompl_interface::ModelBasedPlanningContext::solve(planning_interface::MotionPlanDetailedResponse& res)
 {
-  if (solve(request_.allowed_planning_time, request_.num_planning_attempts))
+  moveit_msgs::MoveItErrorCodes moveit_result = solve(request_.allowed_planning_time, request_.num_planning_attempts);
+  if (moveit_result.val == moveit_msgs::MoveItErrorCodes::SUCCESS)
```

Comment:
Why introducing a new variable here? Can't you use `res.error_code_`?
```suggestion
  res.error_code_.val = solve(request_.allowed_planning_time, request_.num_planning_attempts);
  if (res.error_code_.val == moveit_msgs::MoveItErrorCodes::SUCCESS)
```